Zend/zend_builtin_functions.c

```diff
@@ -592,7 +592,7 @@ ZEND_FUNCTION(define)
 	/* non persistent */
 	ZEND_CONSTANT_SET_FLAGS(&c, 0, PHP_USER_CONSTANT);
 	c.name = zend_string_copy(name);
-	if (zend_register_constant(&c) == SUCCESS) {
+	if (zend_register_constant(&c) != NULL) {
 		RETURN_TRUE;
 	} else {
 		RETURN_FALSE;
```

Comment:
I guess this could also just be `RETURN_BOOL()` but that's a follow-up as I see many other places in this file where it would make sense to do that.